### ISSUE TITLE
Types for `shopify.on(...)` appear to be incorrect

### ISSUE THREAD

**Issue by TomVance** [MONEI/Shopify-api-node#400]
Hey 👋 

It looks like there is a typing issue for the `on` events for changes to call limits. It appears this was looked at with regards to #399 however I'm not sure that worked as expected.

The type is now coming through as:
```
(property) on: (event: never, callback: ((limits: ShopifyClient.ICallLimits) => void) & ((limits: ShopifyClient.ICallGraphqlLimits) => void)) => ShopifyClient
```

I think this is due to the attempt to overload with a union type which doesn't quite work.

This looks to work for us:

```typescript
  on(event: 'callLimits', callback: (limits: Shopify.ICallLimits) => void): Shopify
  on(event: 'callGraphqlLimits', callback: (limits: Shopify.ICallGraphqlLimits) => void): Shopify
  on(event: string, callback: (limits: any) => void): Shopify
```

**Comment by lpinca** [comment]
I get back TS2717 with those definitions. I'm not sure how to fix this as I don't use TypeScript. Pull request is welcome.

**Comment by lpinca** [comment]
Nevermind, it should be fixed on `shopify-api-node@3.4.2`. Thanks for the heads up.

**Comment by TomVance** [comment]
Nice one @lpinca thanks for the speedy resolution.
